internal/controller/managedcluster_controller.go

```diff
@@ -303,24 +309,138 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, l logr.Logger, ma
 		requeue, err := r.setStatusFromClusterStatus(ctx, l, managedCluster)
 		if err != nil {
 			if requeue {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+				return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, err
 			}
 
 			return ctrl.Result{}, err
 		}
 
 		if requeue {
-			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+			return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
 		}
 
 		if !fluxconditions.IsReady(hr) {
-			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+			return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
 		}
+
+		return r.updateServices(ctx, managedCluster)
 	}
 
 	return ctrl.Result{}, nil
 }
 
+// updateServices reconciles services provided in ManagedCluster.Spec.Services.
+// TODO(https://github.com/Mirantis/hmc/issues/361): Set status to ManagedCluster object at appropriate places.
+func (r *ManagedClusterReconciler) updateServices(ctx context.Context, mc *hmc.ManagedCluster) (ctrl.Result, error) {
+	l := log.FromContext(ctx).WithValues("ManagedClusterController", fmt.Sprintf("%s/%s", mc.Namespace, mc.Name))
+	opts := []sveltos.HelmChartOpts{}
+
+	// NOTE: The Profile object will be updated with no helm
+	// charts if len(mc.Spec.Services) == 0. This will result in the
+	// helm charts being uninstalled on matching clusters if
+	// Profile originally had len(m.Spec.Sevices) > 0.
+	for _, svc := range mc.Spec.Services {
+		if svc.Disable {
+			l.Info(fmt.Sprintf("Skip adding Template (%s) to Profile (%s) because Disable=true", svc.Template, mc.Name))
+			continue
+		}
+
+		tmpl := &hmc.ServiceTemplate{}
+		tmplRef := types.NamespacedName{Name: svc.Template, Namespace: mc.Namespace}
+		if err := r.Get(ctx, tmplRef, tmpl); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get Template (%s): %w", tmplRef.String(), err)
+		}
+
+		source, err := r.getServiceTemplateSource(ctx, tmpl)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not get repository url: %w", err)
+		}
+
+		opts = append(opts, sveltos.HelmChartOpts{
```

Comment:
although this works well with templates from our registry, it won't be the case for other flux sources (e.g. git repo source or registries under auth).

it would be nice to use chart urls hosted by source-controller itself. but only if sveltos downloads charts on the mothership cluster. could you please check?

Reply:
I have created the following ticket to implement this in future: https://github.com/Mirantis/hmc/issues/413. This PR as is will work for dev and demo purposes.